url8867e6a4e2_cyberpython_JAwesomeChart_tgz-pJ8-com_gmigdos_jawesomechart_renderers_twodimensional_horizontal_ColumnChartRendererJ8/src/com/gmigdos/jawesomechart/renderers/twodimensional/horizontal/LineChartRenderer.java

```diff
@@ -0,0 +1,283 @@
+import com.gmigdos.jawesomechart.util.Context2D;
+import com.gmigdos.jawesomechart.util.StringsProvider;
+import com.gmigdos.jawesomechart.util.Utilities;
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Stroke;
+import java.awt.geom.Ellipse2D;
+import java.awt.geom.Path2D;
+import java.io.File;
+
+/**
+ *
+ * @author Georgios Migdos <cyberpython@gmail.com>
+ */
+public class LineChartRenderer extends Basic2DHorizontalChartRenderer{
+    
+    private Stroke lineStroke;
+    private float lineOpacity;
+    private float pointOpacity;
+    private double pointRadius;
+    private boolean drawLines;
+    private boolean drawPoints;
+
+    public LineChartRenderer() {
+        this.lineStroke = new BasicStroke(4.0f, BasicStroke.CAP_ROUND, BasicStroke.JOIN_MITER);
+        this.lineOpacity = 1.0f;
+        this.pointOpacity = 1.0f;
+        this.pointRadius = 6;
+        this.drawLines = true;
+        this.drawPoints = false;
+    }
+    
+    /**
+     * @return the lineStroke
+     */
+    public Stroke getLineStroke() {
+        return lineStroke;
+    }
+
+    /**
+     * @param lineStroke the lineStroke to set
+     */
+    public void setLineStroke(Stroke lineStroke) {
+        this.lineStroke = lineStroke;
+    }
+
+    /**
+     * @return the lineOpacity
+     */
+    public float getLineOpacity() {
+        return lineOpacity;
+    }
+
+    /**
+     * @param lineOpacity the lineOpacity to set
+     */
+    public void setLineOpacity(float lineOpacity) {
+        this.lineOpacity = lineOpacity;
+    }
+
+    /**
+     * @return the pointOpacity
+     */
+    public float getPointOpacity() {
+        return pointOpacity;
+    }
+
+    /**
+     * @param pointOpacity the pointOpacity to set
+     */
+    public void setPointOpacity(float pointOpacity) {
+        this.pointOpacity = pointOpacity;
+    }
+
+    /**
+     * @return the pointRadius
+     */
+    public double getPointRadius() {
+        return pointRadius;
+    }
+
+    /**
+     * @param pointRadius the pointRadius to set
+     */
+    public void setPointRadius(double pointRadius) {
+        this.pointRadius = pointRadius;
+    }
+
+    /**
+     * @return the drawLines
+     */
+    public boolean isDrawLinesOn() {
+        return drawLines;
+    }
+
+    /**
+     * @param drawLines the drawLines to set
+     */
+    public void setDrawLines(boolean drawLines) {
+        this.drawLines = drawLines;
+    }
+
+    /**
+     * @return the drawPoints
+     */
+    public boolean isDrawPointsOn() {
+        return drawPoints;
+    }
+
+    /**
+     * @param drawPoints the drawPoints to set
+     */
+    public void setDrawPoints(boolean drawPoints) {
+        this.drawPoints = drawPoints;
+    }
+    
+
+    @Override
+    public String getHumanReadableName() {
+        return StringsProvider.CHART_RENDERER_LINE;
+    }
+
+    @Override
+    public void draw(Context2D context, DataSeriesList series, Labels labels) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.